sequence_processing_pipeline/tests/test_ConvertJob.py

```diff
@@ -995,6 +996,199 @@ def test_parse_sample_sheet(self):
+            job.copy_sequences(sample_name,
+                               source_project,
+                               not_a_project,
+                               copy_all_replicates=False)
+
+        with self.assertRaisesRegex(ValueError, "source 'Feist_11661' and "
+                                                "destination 'Feist_11661' "
+                                                "projects are the same"):
+            job.copy_sequences(sample_name,
+                               source_project,
+                               source_project,
+                               copy_all_replicates=False)
+
+    def test_copy_sequences_success(self):
+        run_dir = self.base_path('211021_A00000_0000_SAMPLE')
+        qiita_id = 'abcdabcdabcdabcdabcdabcdabcdabcd'
+
+        job = ConvertJob(run_dir, self.good_output_path,
+                         self.sample_sheet_path, 'qiita', 1, 16, 1440, '10gb',
+                         'tests/bin/bcl-convert', [], qiita_id)
+
+        sample_name = 'CDPH-SAL.Salmonella.Typhi.MDL-154'
+        source_project = 'Feist_11661'
+        dest_project = 'NYU_BMS_Melanoma_13059'
+        projects = ['NYU_BMS_Melanoma_13059', 'Gerwick_6123', 'Feist_11661']
+
+        # since we can't perform a real run, let's manually create a fake
+        # fastq file and project directories in the 'output_dir' directory and
+        # manually call job._get_sample_sheet_info() to obtain all of the
+        # metadata needed to copy a sequence from one project into another.
+
+        for some_project in projects:
+            # fake the fastq file directories in ConvertJob, one for each
+            # project defined in the sample-sheet.
+            makedirs(join(self.good_output_path, 'ConvertJob', some_project))
+
+        # fake a fastq file in the 'Feist_11661' directory for the purposes of
+        # copying it into the 'NYU_BMS_Melanoma_13059' project.
+        with open(join(self.good_output_path, 'ConvertJob', source_project,
+                       'CDPH-SAL_Salmonella_Typhi_MDL-154_S1_L001_R1_001.'
+                       'fastq.gz'), 'w') as f:
+            f.write("Hello World!\n")
+
+        # manually call the functionality that reads the sample-sheet and
+        # attempts to associate samples with the fastq files generated by
+        # bcl-convert.
+        job._get_sample_sheet_info()
+
+        # copy all fastq files associated w/
+        # 'CDPH-SAL.Salmonella.Typhi.MDL-154' from 'Feist_11661' to
+        # 'NYU_BMS_Melanoma_13059'. 'Gerwick_6123' should remain empty; the
+        # code shouldn't copy anything into that project.
+        job.copy_sequences(sample_name, source_project, dest_project,
+                           copy_all_replicates=False)
+
+        sample_info = job.info[source_project]['samples'][sample_name]
+        # {'Sample_Name': 'CDPH-SAL.Salmonella.Typhi.MDL-154',
+        #  'Sample_ID': 'CDPH-SAL_Salmonella_Typhi_MDL-154',
+        #  'matching_files': []}
```

Comment:
Err, `matching_files` won't really be `[]` here, right?

Reply:
Thanks for the catch Amanda! No it won't. I'll remove the entire comment.